File: packages/ogdf.js/src/layouts/energybased/FastMultipoleEmbedder.js
import { PARAMETER_TYPE } from '../../utils/parameters'

export default {
    defaultEdgeLength: {
        type: PARAMETER_TYPE.DOUBLE,
        range: [0, Infinity],
        default: 1.0
    },
    defaultNodeSize: {
        type: PARAMETER_TYPE.DOUBLE,
        range: [0, Infinity],
        default: 1.0
    },
    multipolePrec: {
        type: PARAMETER_TYPE.INT,
        range: [0, Infinity],
        default: 5
    },
    numberOfThreads: {
        type: PARAMETER_TYPE.INT,
        range: [0, Infinity],
        default: 0
    },
    numIterations: {
        type: PARAMETER_TYPE.INT,
        range: [0, Infinity],
        default: 100
    },
    randomize: {
        type: PARAMETER_TYPE.BOOL,
        range: [true, false],
        default: true
    }
}

File: packages/ogdf.js/src/entry/energybased/FastMultipoleEmbedder.cpp
#include <ogdf/energybased/FastMultipoleEmbedder.h>
#include "../main.h"

EM_PORT_API(FastMultipoleEmbedder *)
LayoutModule_FastMultipoleEmbedder(float defaultEdgeLength, float defaultNodeSize, uint32_t multipolePrec, uint32_t numberOfThreads, uint32_t numIterations, bool randomize)
{
    //LayoutModule
    FastMultipoleEmbedder *model = new FastMultipoleEmbedder();

    model->setDefaultEdgeLength(defaultEdgeLength);
    model->setDefaultNodeSize(defaultNodeSize);
    model->setMultipolePrec(multipolePrec);
    model->setNumberOfThreads(numberOfThreads);
    model->setNumIterations(numIterations);
    model->setRandomize(randomize);

    return model;
}
